### DISCUSSION TITLE
Fine-tuning MACE with different functionals

### DISCUSSION THREAD

**Discussion by hhlim12** [ACEsuit/mace#1151] [Q&A] [answered]
Dear developers, 

I have a question about fine-tuning the foundation model.
Can we fine-tune a model that was trained with one DFT functional (e.g., PBE) using additional data from another functional (e.g., RPBE, SCAN)? If so, I wonder how it works and what kind of adjustment I should make to the following script? 

I noticed a user tried with RPBE (https://github.com/ACEsuit/mace/discussions/1020), but I'm not sure it is the correct procedure..

Do you have any reference explaining how this fine-tuning works? 
Is it kind of delta-learning in which the model learn the energy/force difference?

I would appreciate it if you could provide me with some explanations.
Thank you in advance..

**Comment by gabor1sg** [comment]
The reason this work is because (unless you set it up otherwise), MACE learns the _atomization_ energy, the difference between your system energy and the energy of the equivalent number of isolated atoms. When you switch functionals (or pseudo potentials, or other settings), the absolute energy can change a lot, while the atomisation energy changes a little, and the fine-tuning adapts to the new atomisation energy function. This is why you need to give new isolated atom energies ("E0s") when you fine-tune. IT's not delta learning, the weights are adjusted to match the new data. We typically use so-called multi-head learning, so that very little new data is sufficient, because the model also tries to keep the accuracy for the old data (as represented by some chosen subset).

**Reply by hhlim12** [reply]
Thank you for your kind explanation.
I'm still a bit confused about how it works, but I will try this out.
The multi-head learning that you mentioned is it the one that you explain here? (https://mace-docs.readthedocs.io/en/latest/guide/multihead_finetuning.html#tips-for-successful-multihead-finetuning)?
So does the 'old data' that you mention refer to the replay dataset?

Please allow me to ask one more question. 
Do you have a suggestion for choosing the finetuning set? 
In the case of active learning (say, using GP uncertainty or the committee method), we try to generate and select new data from an MD run.
But in this case, can we just take a subset of the replay dataset and recalculate with a different functional?

**Reply by gabor1** [reply, marked as answer]
yes, that's the multi-head learning. the "old data" is the replay yes. you could indeed just recalculate old configs with your new functional. but I think it's probably better to run some MD of your actual system, and take samples from that. you can try to be fancy with active learning but we have found that so little data is needed for this kind of fine-tuning, that often just taking random samples from MD is sufficient.

**Reply by hhlim12** [reply]
I got it. Thanks a lot!
